### PR TITLE
Added small section to Editor Components

### PR DESCRIPTION
**Pull request by i582**
The section describes convenient `EditorTextField` subclasses.

### CODE REVIEW

File: topics/user_interface_components/editor_components.md [hunk 1]
```diff
@@ -20,6 +20,23 @@ Several commonly needed customization implementations exist, including:
...
+it provides a more Kotlin UI DSL friendly way of creating with the ability to set a default value on creation.
+
+If you want to add autocompletion to the editor, then consider using
+[`TextFieldWithCompletion`](upsource:///platform/platform-impl/src/com/intellij/util/textCompletion/TextFieldWithCompletion.java).
+The constructor takes as an argument a class that implements
+[`TextCompletionProvider`](upsource:///platform/platform-impl/src/com/intellij/util/textCompletion/TextCompletionProvider.java)
+to provide autocompletion variants.
+Use a class
+[`TextFieldCompletionProvider`](upsource:///platform/lang-impl/src/com/intellij/util/TextFieldCompletionProvider.java)
+to create your own provider. For this, override the `addCompletionVariants()` method in which put all the completion variants using the `CompletionResultSet.addElement()` method.
```
Comment (YannCebron) [inline topics/user_interface_components/editor_components.md:36]: .. override `addCompletionVariants()` and add completion variants using `CompletionResultSet.addElement()`.

File: topics/user_interface_components/editor_components.md [hunk 2]
```diff
@@ -20,6 +20,23 @@ Several commonly needed customization implementations exist, including:
...
+
```
Comment (YannCebron) [inline topics/user_interface_components/editor_components.md:37]: worth mentioning TextFieldCompletionProviderDumbAware to allow completion even during indexing

**Review by YannCebron — COMMENTED** [review]
Nice, thanks a lot!

File: topics/intro/content_updates.md [hunk 3]
```diff
@@ -14,6 +14,9 @@ See [GitHub Changelog](https://github.com/JetBrains/intellij-sdk-docs/commits/ma
 
 ### May-22
 
+Editor Components
+: Add a small section to [](editor_components.md) describing convenient `EditorTextField` subclasses.
```
[the remark this answers is not in the document]
Comment (i582) [reply topics/intro/content_updates.md]: Yep, fixed.

File: topics/user_interface_components/editor_components.md [hunk 4]
```diff
@@ -20,6 +20,23 @@ Several commonly needed customization implementations exist, including:
 - [`HorizontalScrollBarEditorCustomization`](upsource:///platform/platform-impl/src/com/intellij/ui/HorizontalScrollBarEditorCustomization.java) to turn on/off horizontal scrollbar
 - [`ErrorStripeEditorCustomization`](upsource:///platform/platform-impl/src/com/intellij/ui/ErrorStripeEditorCustomization.java) to turn on/off error stripes on right
 
+`EditorTextField` has a number of subclasses that can be used as needed for additional features.
+
+If you want to use an editor as an input field in a dialog box, then consider using
+[`LanguageTextField`](upsource:///platform/platform-impl/src/com/intellij/ui/LanguageTextField.java),
+it provides a more Kotlin UI DSL friendly way of creating with the ability to set a default value on creation.
```
[the remark this answers is not in the document]
Comment (i582) [reply topics/user_interface_components/editor_components.md]: Fixed.

File: topics/user_interface_components/editor_components.md [hunk 5]
```diff
@@ -20,6 +20,23 @@ Several commonly needed customization implementations exist, including:
...
+
+If you want to add autocompletion to the editor, then consider using
```
[the remark this answers is not in the document]
Comment (i582) [reply topics/user_interface_components/editor_components.md]: Fixed.

File: topics/user_interface_components/editor_components.md [hunk 6]
```diff
@@ -20,6 +20,23 @@ Several commonly needed customization implementations exist, including:
...
+[`TextFieldWithCompletion`](upsource:///platform/platform-impl/src/com/intellij/util/textCompletion/TextFieldWithCompletion.java).
+The constructor takes as an argument a class that implements
+[`TextCompletionProvider`](upsource:///platform/platform-impl/src/com/intellij/util/textCompletion/TextCompletionProvider.java)
+to provide autocompletion variants.
+Use a class
```
[the remark this answers is not in the document]
Comment (i582) [reply topics/user_interface_components/editor_components.md]: Fixed.

File: topics/user_interface_components/editor_components.md (hunk 1 above)
Comment (i582) [reply topics/user_interface_components/editor_components.md]: Fixed.

File: topics/user_interface_components/editor_components.md (hunk 2 above)
Comment (i582) [reply topics/user_interface_components/editor_components.md:37]: Added.

**Comment by i582** [comment]
Thanks for the review!

**Review by YannCebron — APPROVED** [review]

**Pull request merged**